tests/utils/test_environment.py

```diff
@@ -50,3 +51,27 @@ def test_mlflow_conda_env_includes_pip_dependencies_and_pip_is_specified(pip_spe
         assert conda_dep in env["dependencies"]
     assert pip_specification in env["dependencies"]
     assert env["dependencies"].count("pip") == (2 if pip_specification == "pip" else 1)
+
+
+def _normalize_version(version):
+    """
+    Normalizes a package version (e.g. '1.5.0-dev0' -> '1.5.0.dev0')
+    """
+    from packaging.version import Version
+
+    return str(Version(version))
```

Comment:
The `METADATA` file of `xgboost 1.5.0-dev.0` (note `-` before `dev`) looks like this:


```diff
  Metadata-Version: 2.1
  Name: xgboost
+ Version: 1.5.0.dev0
  Summary: XGBoost Python Package
  Home-page: https://github.com/dmlc/xgboost
  Maintainer: Hyunsu Cho
  Maintainer-email: chohyu01@cs.washington.edu
  ...
```